### PR TITLE
HW3

### PR DESCRIPTION
**Pull request by watchbot23**

### CODE REVIEW

File: GDM.HW3.ConditionalsMethods/Program.cs [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+        //Task 5
+        static int[] ReccursiveQuickArraySort(int[] array, int firstArrayIndex, int lastArrayIndex)
+        {
+            int f = firstArrayIndex;
+            int l = lastArrayIndex;
+            int mid = array[(f + l) / 2];
+            do
+            {
+                while (array[f] < mid)
+                    ++f;
```
Comment (LiDmYr) [inline GDM.HW3.ConditionalsMethods/Program.cs]: Please use brackets with _while_, _for_, _if_, etc.

File: GDM.HW3.ConditionalsMethods/Program.cs [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+            {
+                if (array[i] == veriable && arrayIndex != i)
+                {
+                    temp = true;
+                    return temp;
+                }
+            }
+            return temp;
+        }
+        static void Main(string[] args)
```
Comment (LiDmYr) [inline GDM.HW3.ConditionalsMethods/Program.cs:12]: Main is a main method - it should be the first among all methods

File: GDM.HW3.ConditionalsMethods/Program.cs [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+            Random randomizer = new Random();
+
+            //Task 1
+            string task1 = "Task 1.";
+            var eggsNumberInBox = 10;
+            var eggs = 0;
+            while (eggs < eggsNumberInBox)
+            {
+                eggs++;
+
```
Comment (LiDmYr) [inline GDM.HW3.ConditionalsMethods/Program.cs]: What is an aim of this empty line ?

File: GDM.HW3.ConditionalsMethods/Program.cs [hunk 4]
```diff
@@ -0,0 +1,315 @@
...
+            }
+            Console.WriteLine(task1 + $"1. Количество яиц в лотке {eggs} шт.");
+            int eggs2 = 10;
+            int count = 0;
+            do
+            {
+                eggs--;
+                count++;
+
```
Comment (LiDmYr) [inline GDM.HW3.ConditionalsMethods/Program.cs]: What is an aim of this empty line ?

File: GDM.HW3.ConditionalsMethods/Program.cs [hunk 5]
```diff
@@ -0,0 +1,315 @@
...
+            double fac = randomizer.Next(0, 20);
+            Console.WriteLine(task2 + $"Факториал({fac}) = {Factorial(fac)}");
+
+            //Task 3
+            string task3 = "Task 3. ";
+            int fib = randomizer.Next(0, 30);
+            Console.WriteLine(task3 + $"Последовательностью чисел Фибоначчи для {fib} является:");
+            for (int i = 0; i < fib; i++)
+            {
+                Console.Write(Fibonachi(i) + " ");
```
Comment (LiDmYr) [inline GDM.HW3.ConditionalsMethods/Program.cs]: You repeated the same calculation a lot of times. Try to call only one time this function.

File: GDM.HW3.ConditionalsMethods/Program.cs [hunk 6]
```diff
@@ -0,0 +1,315 @@
...
+using System.Threading.Tasks;
+
+namespace GDM.HW3.ConditionalsMethods
+{
+    class Program
+    {
+        //Task 2
+        static double Factorial(double fac)
+        {
+            if (fac == 0)
```
Comment (LiDmYr) [inline GDM.HW3.ConditionalsMethods/Program.cs]: use ternary operator + handle a case when 'fac = -1', etc

File: GDM.HW3.ConditionalsMethods/Program.cs [hunk 7]
```diff
@@ -0,0 +1,315 @@
...
+        //Task 3
+        static int Fibonachi(int n)
+        {
+            return n > 1 ? Fibonachi(n - 1) + Fibonachi(n - 2) : n;
+        }
+
+        //Task 4
+        static int DaysNumberInMonth(int month, int year)
+        {
+            if (month == 2 && year % 4 == 0 && year % 100 != 0 || month == 2 && year % 100 == 0)
```
Comment (LiDmYr) [inline GDM.HW3.ConditionalsMethods/Program.cs]: Use if(...){...}else if(...){...}etc. - of course you use 'return' and it's not possible to go to another if(...) but it looks "strange"

File: GDM.HW3.ConditionalsMethods/Program.cs [hunk 8]
```diff
@@ -0,0 +1,315 @@
...
+            {
+                return 28;
+            }
+            if (month % 2 == 1 && month < 7 || month > 8 && month % 2 == 1)
+            {
+                return 30;
+            }
+            else
+            {
+                return 31;
```
Comment (LiDmYr) [inline GDM.HW3.ConditionalsMethods/Program.cs]: Try to assign variable and return one time

File: GDM.HW3.ConditionalsMethods/Program.cs [hunk 9]
```diff
@@ -0,0 +1,315 @@
...
+            }
+        }
+
+        //Task 5
+        static int[] ReccursiveQuickArraySort(int[] array, int firstArrayIndex, int lastArrayIndex)
```
Comment (LiDmYr) [inline GDM.HW3.ConditionalsMethods/Program.cs]: Your algorithm is faster in 5-10 times than my example in github :) GREAT JOB:)

File: GDM.HW3.ConditionalsMethods/Program.cs [hunk 10]
```diff
@@ -0,0 +1,315 @@
...
+        {
+            int f = firstArrayIndex;
```
Comment (LiDmYr) [inline GDM.HW3.ConditionalsMethods/Program.cs]: Try to use human readable variables names

File: GDM.HW3.ConditionalsMethods/Program.cs [hunk 11]
```diff
@@ -0,0 +1,315 @@
...
+
+        //Task 6
+        static int[] PutZeroBetweenMimMaxValueArray(int[] array)
+        {
+            int[] tempArray = (int[])array.Clone();
+            int[] sortedRandomArray = ReccursiveQuickArraySort(tempArray, 0, tempArray.Length - 1);
+            int min = 0;
+            int max = 0;
+
+            for (int i = 0; i < array.Length; i++)
```
Comment (LiDmYr) [inline GDM.HW3.ConditionalsMethods/Program.cs]: Why do use 2 time the same for loop on 95 + 102 lines?

File: GDM.HW3.ConditionalsMethods/Program.cs [hunk 12]
```diff
@@ -0,0 +1,315 @@
...
+                }
+            }
+            for (int i = 0; i < array.Length; i++)
+            {
+                if (array[i] == sortedRandomArray[sortedRandomArray.Length - 1])
+                {
+                    max = i;
+                }
+            }
+            if (min > max)
```
Comment (LiDmYr) [inline GDM.HW3.ConditionalsMethods/Program.cs]: Rewrite 109-122 lines with only one _if(){}else{}_ and only one _for_

File: GDM.HW3.ConditionalsMethods/Program.cs [hunk 13]
```diff
@@ -0,0 +1,315 @@
...
+        //Task 8
+        static bool IsDupl(int[] array, int veriable, int arrayIndex)
+        {
+            bool temp = false;
+            for (int i = 0; i < array.Length; i++)
+            {
+                if (array[i] == veriable && arrayIndex != i)
+                {
+                    temp = true;
+                    return temp;
```
Comment (LiDmYr) [inline GDM.HW3.ConditionalsMethods/Program.cs]: You don't need to **return temp** (you already have this on 138 line) here - just break for.

File: GDM.HW3.ConditionalsMethods/Program.cs [hunk 14]
```diff
@@ -0,0 +1,315 @@
...
+                    array[i] = 0;
+                }
+            }
+            return array;
+        }
+
+        //Task 8
+        static bool IsDupl(int[] array, int veriable, int arrayIndex)
+        {
+            bool temp = false;
```
Comment (LiDmYr) [inline GDM.HW3.ConditionalsMethods/Program.cs]: Write another name - **_isSomethingHappened_**

**Review by LiDmYr — COMMENTED** [review]
I would like you implement my notes/comments

File: GDM.HW3.ConditionalsMethods/Program.cs (hunk 11 above)
Comment (watchbot23) [reply GDM.HW3.ConditionalsMethods/Program.cs]: cos I don't know if the max value has bigger index than min one or not.

File: GDM.HW3.ConditionalsMethods/Program.cs (hunk 9 above)
Comment (watchbot23) [reply GDM.HW3.ConditionalsMethods/Program.cs]: wow)

File: GDM.HW3.ConditionalsMethods/Program.cs (hunk 13 above)
Comment (watchbot23) [reply GDM.HW3.ConditionalsMethods/Program.cs]: oops. I know. Just fogot to delete it.

File: GDM.HW3.ConditionalsMethods/Program.cs [hunk 15]
```diff
@@ -0,0 +1,341 @@
...
+                    {
+                        Console.Write(arraysArray[i][j] + "\n");
+                    }
+                }
+            }
+        }
+        //Task 8
+        static bool IsArrayVeriableDuplicated(int[] array, int veriable, int arrayIndex)
+        {
+            bool temp = false;
```
Comment (LiDmYr) [inline GDM.HW3.ConditionalsMethods/Program.cs:330]: please change the name

File: GDM.HW3.ConditionalsMethods/Program.cs [hunk 16]
```diff
@@ -0,0 +1,341 @@
...
+            for (int i = 0; i < array.Length; i++)
+            {
+                if (array[i] == veriable && arrayIndex != i)
+                {
+                    temp = true;
+                }
```
Comment (LiDmYr) [inline GDM.HW3.ConditionalsMethods/Program.cs:336]: why do you need to continue for loop if you already have the 'true' ?

**Review by LiDmYr — COMMENTED** [review]
Actually not all changes which I asked you made

**Pull request merged**